### ISSUE TITLE
Page reloading query

### ISSUE THREAD

**Issue by chrisdwheatley** [geelen/jspm-server#18]
I'm struggling with my application still wanting to reload even though I seem to have everything in place to allow modules to be replaced without a full page refresh and was hoping to get some insight/troubleshooting help.

AFAIK I've got everything set up correctly, I've got `System.trace = true` in `index.html` and within each component I'm exporting `__hotReload = true`. Things are working correctly, hot replacement happens with no page refresh, but then the page refreshes straight after that. See screenshot & gif below.

![screen shot 2015-06-29 at 11 16 44](https://cloud.githubusercontent.com/assets/894092/8406685/963d457c-1e56-11e5-89e1-a00cc574400c.png)

![jspm-server](https://cloud.githubusercontent.com/assets/894092/8406568/8ae4135a-1e55-11e5-83b5-0af72ebccf5f.gif)

Is it the fact that jspm-server is watching my whole repository and it's seeing files in the `src` directory change which it can't handle and falls back to refreshing (that's what the console warning suggests to me anyway)?

The gif should give some insight into my setup, file structure etc... but I'm not sure if that's the optimal way, it'd be interesting to know how you've got things set up in your dev env @geelen. Perhaps it's a workflow thing and I need to change the setup a bit to accomodate.

If you want to see the code, config, gulpfile I'm using you can see the [repo here](https://github.com/plus-n-boots/plus-n-boots.github.io).

**Comment by kevrom** [comment]
It definitely looks like the build is triggering a hard reload, since Babel is transpiling your classes into the ES5 syntax of creating classes.  Hot reloading requires the ES6 class syntax, so that transpile step is breaking it.  Have you tried just letting JSPM handle the build process implicitly instead of using your gulpfile?  There's really no reason to have gulp transpiling and using jspm simultaneously.  Just `System.import('src/index')` instead and it should work.

You also have the option use https://github.com/geelen/jspm-loader-jsx to automatically add the `__hotReload = true` to files that fit the criteria (class and filenames match, have to be jsx files I believe, have to import the files like `import something from './MyClass.jsx!'`)

It looks like there's a way to make the whole process a lot cleaner using the beta System.js meta configurations, but I think the apis changed a bit and broke some things so I'm working them out currently.

**Comment by geelen** [comment]
Right, so I think what is happening is that gulp & jspm-server aren't playing nicely together. You're changing a JSX file, _both_ gulp and jspm-server are detecting it, jspm-server can't handle it so it starts a reload, _then_ gulp generates a new JS file which is live-injected. But the reload has already been queued, and so it reloads.

Try using `jspm-server --ignore-exts="jsx"` and see what happens. Alternatively, switch to [jspm-loader-jsx](https://github.com/geelen/jspm-loader-jsx) and remove your JSX stuff from your gulpfile.

Thanks for posting so much detail!
